Sources/SecureXPC/Server/SecureMessageAcceptor.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  SecureMessageAcceptor.swift
+//  SecureXPC
+//
+//  Created by Josh Kaplan on 2022-01-06
+//
+
+import Foundation
+
+/// Accepts messages which meet the provided code signing requirements.
+///
+/// Uses undocumented functionality prior to macOS 11.
+internal struct SecureMessageAcceptor {
+    /// At least one of these code signing requirements must be met in order for the message to be accepted
+    internal let requirements: [SecRequirement]
+    
+    func acceptMessage(connection: xpc_connection_t, message: xpc_object_t) -> Bool {
+        // Get the code representing the client
+        var code: SecCode?
+        if #available(macOS 11, *) { // publicly documented, but only available since macOS 11
+            SecCodeCreateWithXPCMessage(message, SecCSFlags(), &code)
+        } else { // private undocumented function: xpc_connection_get_audit_token, available on prior versions of macOS
+            if var auditToken = xpc_connection_get_audit_token(connection) {
```

Comment:
If you add an `else` case, then `code` will be initialized in all code paths, removing the need for it to be optional.

Better yet, you should just flip this with a guard as well: `guard var auditToken = ... else { return false }`

Reply:
My understanding is `code` needs to be optional because both `SecCodeCreateWithXPCMessage` and `SecCodeCopyGuestWithAttributes` expect a parameter of type `UnsafeMutablePointer<SecCode?>` which can't be satisfied with a non-optional type. Is there a way to make this work?

Incorporated feedback on second point about guard statement

Reply:
AH I didn't know that was the expected param type. Usually optional promotion happens, but not in the case of using `&` to generate a temporary pointer. You'd have to check, maybe it does do optional promotion in that case too, IDR.

Reply:
Unfortunately it's not allowed. Xcode outputs:
> Cannot convert value of type 'UnsafeMutablePointer<SecCode>' to expected argument type 'UnsafeMutablePointer<SecCode?>'